### PR TITLE
mime_type,photo_info and video_info parsing

### PR DESCRIPTION
**Pull request by richardtitze**
I have added support for file mime_type, photo_info and video_info parsing to DbxEntry as this was missing and i really this for the project i am working on.

### CODE REVIEW

**Comment by cakoose** [comment]
What are you using "mime_type" for?  We're looking to deprecate the "mime_type" field, which is why it wasn't included in the SDK, so I'd prefer not to include it.

The photo/video info stuff is generally good, but I've included some comments on the commit.

**Comment by richardtitze** [comment]
I would need the mime type for classifying files without having to download them, and without having to rely on the file-extension. I'll remove it for now and do classification based on file-extensions. It would be great if you decided to not deprecate it and include it in the sdk.

**Comment by cakoose** [comment]
Currently, the Dropbox server just picks the mime_type based on the file extension.  This is something the app can do with its own extension-to-mime_type mapping.  Further, if the app is dealing with files from other sources, it's good to have that mapping be consistent no matter where the file comes from.

**Comment by cakoose** [comment]
I don't see any place where you're passing "include_media_info".  Were you ever getting photo/video info without that flag?

If this was just an oversight, don't worry.  I can fix it.  I'm just wondering if the server is returning photo/video info when you don't pass the "include_media_info" flag.

**Comment by richardtitze** [comment]
Yes, the server is returning photo/video info. I have built my own wrapper client, as i'm using the apache http client, and not the one included in the sdk

**Comment by cakoose** [comment]
In my testing, it looks like the server only returns photo/video info when you pass "include_media_info=true".  Is this not the case for you?


BTW, If you want to use the Apache HTTP client library with `DbxClient`, you can do so by implementing the `HttpRequestor` interface and passing that into `DbxRequestConfig`, which then gets passed to `DbxClient`.  Is there another reason you're not using `DbxClient`?  If so, is there something we can fix to make `DbxClient` work for your use-case?

File: src/com/dropbox/core/DbxEntry.java [hunk 1]
```diff
@@ -322,6 +335,137 @@ public int hashCode()
...
+        }
+
+        /**
+         * A geo location
+         */
+        public static class Location {
+            /**
+             * The latitude of the location
+             */
+            public final BigDecimal latitude;
```
Comment (cakoose) [inline src/com/dropbox/core/DbxEntry.java]: Is there a reason you used `BigDecimal` and not just `double`?  Were you seeing issues when parsing it as a `double`?

Comment (richardtitze) [inline src/com/dropbox/core/DbxEntry.java]: No particular reason. I do store gps coordinates as BigDecimal. I would be
ok though to get them as doubles.

**Comment by richardtitze** [comment]
Also in my tests the server returns the info only when i add
"include_media=true" to the request.

I thought about implementing the HttpRequestor interface, but the problem
is that this interface is push-based, whereas the apache library is pull
based. As i'm using only a small subset of the Core API i just went with a
small wrapper class - only about 240 lines of code - which i'll also use to
propagate my own exceptions.

**Comment by cakoose** [comment]
Good to know.  Do you think you can put your client class up in a GitHub Gist?  It would help me figure out how to accommodate your use case better.

**Comment by richardtitze** [comment]
I have created this gist:
https://gist.github.com/richardtitze/c1a48d7811e38cea3a13

**Comment by cakoose** [comment]
Ah, I understand the problem now.  Thanks.

The SDK only makes use of the "push" capability in the `DbxClient.startUploadFile*` methods.  I'm not sure how much they're used.  Maybe they can be removed.

I'll think about it for a bit.  Thanks for the help.

**Pull request closed without merging**